Feature/component data for 3rd party components (#44)
AdaloHQ/docs: website/docs/docs/interactions/component-data.md

---
id: component-data
title: Component Data
---

Component data is exported from your component for use in the rest of a makers app in the form of Magic Text. You can implement Component Data into your component by assigning props the role of `formValue` in the component manifest:

```json
{
  ...
  "props": [
    {
      "name": "exampleFormValue",
      "displayName": "Example Form Value",
      "type": "text",
      "role": "formValue"
    }
  ]
}
```

In the code, `exampleFormValue` will now refer to a `{ value, onChange }` object. 

> `exampleFormValue.onChange` is a function that should be caused as an event handler, it will update `exampleFormValue.value` when called. `exampleFormValue.value` will refer to the value that the component is exporting and can be accessed throughout the app as Magic Text. An additional prop, `exampleFormValue_initial`, will also be passed to the component. This prop will always refer to the initial value of `exampleFormValue.value` and can be useful for purposes such as setting placeholder values.

Reviewer: ```suggestion
When called, `exampleFormValue.onChange` will update `exampleFormValue.value`. `exampleFormValue.value` refers to the value that the component is exporting and can be accessed throughout the app as Magic Text. 
An additional prop, `exampleFormValue_initial`, will also be passed to the component. This prop will always refer to the initial value of `exampleFormValue.value` and can be useful for purposes such as setting placeholder values.
```

---
id: component-data
title: Component Data
---

> Component data is exported from your component for use in the rest of a makers app in the form of Magic Text. You can implement Component Data into your component by assigning props the role of `formValue` in the component manifest:

Reviewer 2: ```suggestion
Besides actions, component data is another way to connect your component to the rest of the Adalo ecosystem. Makers can use component data by accessing it through Magic Text. You can implement Component Data into your component by assigning props the role of `formValue` in the component manifest:
```

Added that first sentence + second sentence edit to mimic the [actions docs](https://developers.adalo.com/docs/interactions/actions)

```json
{
  ...
  "props": [
    {
      "name": "exampleFormValue",
      "displayName": "Example Form Value",
      "type": "text",
      "role": "formValue"
    }
  ]
}
```

In the code, `exampleFormValue` will now refer to a `{ value, onChange, initial }` object. 

> When called, `exampleFormValue.onChange` will update `exampleFormValue.value`. `exampleFormValue.value` refers to the value that the component is exporting and can be accessed throughout the app as Magic Text. Additionally, `exampleFormValue.initial`will always refer to the initial value of `exampleFormValue.value` and can be useful for purposes such as setting placeholder values.

Reviewer 2: ```suggestion
When called, `exampleFormValue.onChange` will update `exampleFormValue.value`. `exampleFormValue.value` refers to the value that the component is exporting and can be accessed throughout the app as Magic Text. Additionally, `exampleFormValue.initial `will always refer to the initial value of `exampleFormValue.value` and can be useful for purposes such as setting placeholder values.
```

 These values can be used in the code like so:

```javascript
exampleComponent = (props)=> {

>     const { exampleFormValue, exampleFormValue_initial } = props

Reviewer: ```suggestion
    const { exampleFormValue } = props
```
